### DISCUSSION TITLE
Question about how Mastodon search interacts with websites

### DISCUSSION THREAD

**Discussion by patrickmcurry** [mastodon/mastodon#21673] [Q&A] [answered]
I have a question about the Mastodon search protocol / flow / API. When I paste a URL into the Mastodon search box, I see a request being made in my web server logs — for that URL and only that URL. But no results are shown in the Mastodon UI. What data is the Mastodon server looking for to decide if there’s a person or post at the URL?

**Comment by nightpool** [comment, marked as answer]
The Mastodon server is looking for an ActivityPub object, using the `application/activity+json` mime type. Please see https://www.w3.org/TR/activitypub/ and https://docs.joinmastodon.org/activitypub for more details.

**Reply by patrickmcurry** [reply]
Thank you, that pointed me in the right direction. I was able to detect the request was asking for application/activity+json, and then respond with the right mime type header AND json. Much appreciated.

**Reply by trwnh** [reply]
Best practice is to use `application/ld+json; profile="http://www.w3.org/ns/activitystreams"` (which servers MUST respond to) instead of `application/activity+json` (which servers SHOULD respond to, but are not guaranteed)

**Reply by patrickmcurry** [reply]
Meaning my web server should use the application/ld+json MIME type when responding, or it should detect that MIME type in the request and respond to it?

**Reply by trwnh** [reply]
@patrickmcurry both -- detect that a request is using `Accept:` with either of those (ld or activity), and respond with a `Content-Type:` of the former (ld)

Essentially, this is the ideal way to request and respond:

```http
GET /endpoint HTTP/1.1
Accept: application/ld+json; profile="http://www.w3.org/ns/activitystreams"

Content-Type: application/ld+json; profile="http://www.w3.org/ns/activitystreams"
{...}
```

But this is also fine:

```http
GET /endpoint HTTP/1.1
Accept: application/activity+json

Content-Type: application/ld+json; profile="http://www.w3.org/ns/activitystreams"
{...}
```
